source/switch/main.cpp

```diff
@@ -667,8 +675,15 @@ int main(int argc, char *argv[]) {
 				free(buffer);
 				mutexUnlock(&emulationLock);
 			} break;
-			case resultSettingsChanged:
+			case resultSaveSettings:
 				applyConfig();
```

Comment:
I haven't tested it but the save settings or cancel settings doesn't seem right to me. When a setting is changed, currently it just won't be applied, but the changed value persists.
Let's suppose you change a setting and cancel then. Then later you open the settings menu again and you probably see that the previously changed setting is still changed.

Reply:
All fine now 👍